### PR TITLE
lua: add 5.4.1 + patch in build() instead of source() + cache cmake + del fPIC if shared + no libm on Macos + fix hooks

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lua/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — DISMISSED** [review]

File: recipes/lua/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
-
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-file(GLOB SOURCE_FILES *.cpp)
-
-
-
-add_executable(${PROJECT_NAME} ${SOURCE_FILES})
+add_executable(${PROJECT_NAME} test_package.cpp)
```
Comment (madebr) [inline recipes/lua/all/test_package/CMakeLists.txt:7]: Do we need to test lua as a c executable as well?

Comment (SpaceIm) [reply recipes/lua/all/test_package/CMakeLists.txt:7]: Yes probably. The C API is tested but in a cpp file.

File: recipes/lua/all/CMakeLists.txt [hunk 2]
```diff
@@ -22,37 +17,27 @@ conan_basic_setup()
...
+${SOURCE_DIR}/src/lapi.c ${SOURCE_DIR}/src/lauxlib.c ${SOURCE_DIR}/src/lbaselib.c ${SOURCE_DIR}/src/lcode.c ${SOURCE_DIR}/src/lcorolib.c
 ${SOURCE_DIR}/src/lctype.c ${SOURCE_DIR}/src/ldblib.c ${SOURCE_DIR}/src/ldebug.c ${SOURCE_DIR}/src/ldo.c ${SOURCE_DIR}/src/ldump.c ${SOURCE_DIR}/src/lfunc.c ${SOURCE_DIR}/src/lgc.c
 ${SOURCE_DIR}/src/linit.c ${SOURCE_DIR}/src/liolib.c ${SOURCE_DIR}/src/llex.c ${SOURCE_DIR}/src/lmathlib.c ${SOURCE_DIR}/src/lmem.c ${SOURCE_DIR}/src/loadlib.c ${SOURCE_DIR}/src/lobject.c
 ${SOURCE_DIR}/src/lopcodes.c ${SOURCE_DIR}/src/loslib.c ${SOURCE_DIR}/src/lparser.c ${SOURCE_DIR}/src/lstate.c ${SOURCE_DIR}/src/lstring.c ${SOURCE_DIR}/src/lstrlib.c
 ${SOURCE_DIR}/src/ltable.c ${SOURCE_DIR}/src/ltablib.c ${SOURCE_DIR}/src/ltm.c ${SOURCE_DIR}/src/lundump.c ${SOURCE_DIR}/src/lutf8lib.c ${SOURCE_DIR}/src/lvm.c ${SOURCE_DIR}/src/lzio.c
 )
 
+IF(LUA_VERSION VERSION_LESS 5.4.0)
+    LIST(APPEND SRC_LIBLUA ${SOURCE_DIR}/src/lbitlib.c)
+ENDIF()
```
Comment (madebr) [inline recipes/lua/all/CMakeLists.txt:33]: Would this be *simpler* and more future proof without needing any special casing of versions?
```suggestion
file(GLOB SRC_LIBLUA "${SOURCE_DIR}/src/*.c")
list(REMOVE_ITEM SRC_LIBLUA "${SOURCE_DIR}/src/lua.c" "${SOURCE_DIR}/src/luac.c")
```

Comment (SpaceIm) [reply recipes/lua/all/CMakeLists.txt]: Good point, done.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**